src/main/java/org/beanpod/switchboard/dto/mapper/EncoderMapper.java

```diff
@@ -1,13 +1,15 @@
 package org.beanpod.switchboard.dto.mapper;
 
 import org.beanpod.switchboard.dto.EncoderDTO;
+import org.beanpod.switchboard.dto.OutputChannelDTO;
 import org.beanpod.switchboard.entity.EncoderEntity;
+import org.mapstruct.Context;
 import org.mapstruct.Mapper;
 import org.mapstruct.factory.Mappers;
 
 import java.util.List;
 
-@Mapper(componentModel = "spring")
+@Mapper(componentModel = "spring", uses = {DeviceMapper.class, OutputChannelMapper.class})
```

Comment:
Previously we didn't use DeviceMapper so when Mapstruct generated the mapper and used internal private method to map de Device of the Encoders. Because we don't have any special mapping for the Device it boils down to pretty much the same code. But in the future if we need to change Device Mapping the changes will propagate to the *coder. For example, we could use the mapper to "reduce" the device into their respective *coder DTO like I did for the Input/Output channel and `Channel`.